c/datasetjson.c

```diff
@@ -79,22 +79,26 @@ int streamDataset(Socket *socket, char *filename, int recordLength, jsonPrinter
   char buffer[bufferSize];
   memset(buffer,0,bufferSize);
   jsonStartArray(jPrinter,"records");
-
   int contentLength = 0;
   int bytesRead = 0;
   if (in) {
-      while (!feof(in)){      
-        bytesRead = fread(buffer,1,recordLength,in);
+    while (!feof(in)){
+      bytesRead = fread(buffer,1,recordLength,in);
+      if (bytesRead > 0){
         memset(buffer+bytesRead,0,recordLength-bytesRead);
         jsonAddString(jPrinter, NULL, buffer);
-        if (bytesRead < 0 && ferror(in)){
-          printf("Error reading DSN=%s, rc=%d\n",filename,bytesRead);
-        }
-        else{
-          contentLength = contentLength + bytesRead;
-        }
+        contentLength = contentLength + bytesRead;
       }
-      fclose(in);
+      else if (bytesRead == 0){
+        printf("Reached end of dataset\n");
```

Comment:
Since we didn't use any way to differentiate between debug & non debug messages, please remove this printf entirely. It is meaningless to the end user and they will see zillions of them.

Reply:
I've updated this in the latest commit